pkg/agent/entrypoint.go

```diff
@@ -79,18 +75,32 @@ func (r MainRunner) Run(logger *zap.Logger, ctx context.Context) error {
 		}
 	}
 
-	logger.Info("Entering main loop")
-	for {
-		event, err := vmEventQueue.Wait(ctx)
-		if err != nil {
-			if ctx.Err() != nil {
-				// treat context canceled as a "normal" exit (because it is)
-				return nil
-			}
+	mc, err := billing.NewMetricsCollector(ctx, logger, &r.Config.Billing)
+	if err != nil {
+		return fmt.Errorf("error starting billing metrics collector: %w", err)
+	}
 
-			logger.Error("vmEventQueue returned error", zap.Error(err))
-			return err
+	tg := taskgroup.NewGroup(logger)
+	ctx = tg.WithContext(ctx)
+	tg.Go("watch-metrics", func(logger *zap.Logger) error {
```

Comment:
What does the "watch" in "watch-metrics" mean?

Reply:
Yeah, I guess collect metrics should be better.